python/rikai/parquet/writer.py

```diff
@@ -0,0 +1,125 @@
+#  Copyright 2022 Rikai Authors
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+"""Tools to write Rikai format datasets to disk
+"""
+
+import json
+import pyarrow as pa
+import pyarrow.parquet as pq
+from rikai.mixin import ToDict
+from rikai.parquet.dataset import Dataset as RikaiDataset
+import pandas as pd
+
+__all__ = ["df_to_rikai"]
+
+
+def df_to_rikai(
+    df: pd.DataFrame, dest_path: str, schema: "pyspark.sql.types.StructType"
+):
+    """Write the given pandas dataframe to the given location using the
+    specified (pyspark) schema
+
+    Parameters
+    ----------
+    df: pd.DataFrame
+        The DataFrame to write to Rikai format
+    dest_path: str
+        The destination path to write df to
+    schema: StructType
+        The pyspark schema required to convert Rikai types and make the
+        resulting rikai dataset readable by spark
+    """
+    schema_json_str = schema.json()
+    converted_df = conv(df, json.loads(schema_json_str))  # convert Rikai types
+    # The spark schema is required so Spark can read it successfully
+    table = pa.Table.from_pandas(converted_df).replace_schema_metadata(
+        {RikaiDataset.SPARK_PARQUET_ROW_METADATA: schema_json_str}
+    )
+    pq.write_to_dataset(table, dest_path, use_legacy_dataset=False)
```

Comment:
lets also support partitionBy?

Does this overwrite existing data by default? How to control the number of files and etc?